app/script/components/groupVideoGrid.js

```diff
@@ -58,31 +58,59 @@ z.components.GroupVideoGrid = (() => {
       this.streams.subscribe(this.updateGrid.bind(this));
       this.updateGrid(this.streams());
 
+      this.scaleVideos = this.scaleVideos.bind(this, rootElement);
+      // scale videos when the grid is updated (on the next rendering cycle)
+      this.grid.subscribe(() => {
+        window.requestAnimationFrame(this.scaleVideos);
+      });
+
+      // scale the videos when the window is resized
+      window.addEventListener('resize', this.scaleVideos);
+
+      this.rootElement = rootElement;
+
       this.minimized = params.minimized;
     }
 
-    scaleVideos(elements) {
+    scaleVideos(rootElement) {
+      const elements = Array.from(rootElement.querySelectorAll('.group-video-grid__element'));
       elements
         .filter(element => !!element.classList)
         .filter(element => element.classList.contains('group-video-grid__element'))
```

Comment:
No need to filter them by there class name as you already selected only those with that class.

Reply:
Very true